sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/rowBasedHiveUDFs.scala

```diff
@@ -55,6 +55,33 @@ trait GpuRowBasedHiveUDFBase extends GpuRowBasedUserDefinedFunction with HiveIns
 
   override def prettyName: String = name
 
+  /** Handle GpuLiteral and foldable GpuExpression, similar to `toInspector(Expression)` */
+  protected def gpuToInspector(expr: Expression): ObjectInspector = expr match {
+    case GpuLiteral(value, dataType) =>
+      // Convert to CPU literal as possible as we can before going into `toInspector(Expression)`.
+      // Because some optimization will be made for Literal expressions of primitive types.
+      value match {
+        case scalar: ai.rapids.cudf.Scalar =>
+          if (scalar.getType.isNestedType) {
+            // Nested type, so create an inspector from the data type.
+            toInspector(dataType)
+          } else {
+            try {
+              toInspector(Literal.create(GpuScalar.extract(scalar), dataType))
+            } catch {
+              // Unsupported type for extraction, so use the data type way instead.
+              case _: Throwable => toInspector(dataType)
```

Comment:
Normally we don't call our own code and catch exceptions after the fact, rather we check what we know will work and avoid the exceptional path.  This ends up hiding bugs like NPEs and other unexpected exceptions, so I'd rather see us check types up-front or catch the specific exceptions that would be thrown. Or are there exceptions being thrown in `Literal.create` here as well?

Reply:
Nice catch. 
My mistake. Forgot to update this which was filled automatically by the IDE context tool.